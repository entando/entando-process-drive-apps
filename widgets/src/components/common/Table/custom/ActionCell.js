import React from 'react';
import DoneIcon from '@material-ui/icons/Done';
import HowToRegIcon from '@material-ui/icons/HowToReg';
import AssessmentIcon from '@material-ui/icons/Assessment';

import IconMenuButton from 'components/common/IconMenuButton';

const ActionCell = (options, openDiagram) => data => {
  const menuItems = [];

  if (options.showComplete) {
    menuItems.push({
      text: 'Complete',
      icon: <DoneIcon fontSize="small" />,
    });
  }

  if (options.showClaim) {
    menuItems.push({
      text: 'Claim',
      icon: <HowToRegIcon fontSize="small" />,
    });
  }

  menuItems.push({
    text: 'Diagram',
    icon: <AssessmentIcon fontSize="small" />,
    onClick: openDiagram(data.row),
  });

  return <IconMenuButton menuItems={menuItems} />;
};

export default ActionCell;
