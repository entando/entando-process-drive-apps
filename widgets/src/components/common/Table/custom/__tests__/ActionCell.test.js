import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import ActionCell from 'components/common/Table/custom/ActionCell';

describe('<ActionCell />', () => {
  it('renders snapshot correctly', () => {
    const { container } = render(<ActionCell />);

    expect(container).toMatchSnapshot();
  });

  it('shows Claim, Complete and Diagram buttons when button is clicked', () => {
    const { getByText, getByRole } = render(<ActionCell />);

    fireEvent.click(getByRole('button'));

    expect(getByText('Claim')).toBeInTheDocument();
    expect(getByText('Complete')).toBeInTheDocument();
    expect(getByText('Diagram')).toBeInTheDocument();
  });
});
