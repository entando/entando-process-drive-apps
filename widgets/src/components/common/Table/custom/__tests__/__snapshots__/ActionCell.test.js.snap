// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionCell /> renders snapshot correctly 1`] = `<div />`;
